crates/docs/src/lib.rs

```diff
@@ -36,25 +36,50 @@ pub fn generate_docs_html(root_file: PathBuf) {
     fs::create_dir_all(build_dir).expect("TODO gracefully handle being unable to create build dir");
 
     // Copy over the assets
-    fs::write(
-        build_dir.join("search.js"),
-        include_str!("./static/search.js"),
-    )
-    .expect("TODO gracefully handle failing to make the search javascript");
-
-    fs::write(
-        build_dir.join("styles.css"),
-        include_str!("./static/styles.css"),
-    )
-    .expect("TODO gracefully handle failing to make the stylesheet");
-
-    fs::write(
-        build_dir.join("favicon.svg"),
-        include_str!("./static/favicon.svg"),
-    )
-    .expect("TODO gracefully handle failing to make the favicon");
-
-    let template_html = include_str!("./static/index.html")
+    // For debug builds, read assets from fs to speed up build
+    // Otherwise, include as string literal
+
+    // Copy search.js
+    #[cfg(not(debug_assertions))]
+    let search_js = include_str!("./static/search.js");
+
+    #[cfg(debug_assertions)]
+    let search_js = fs::read_to_string("./crates/docs/src/static/search.js").unwrap();
```

Comment:
I think there's some way this can read from the directory relative to this source file - as opposed to `./` which will be relative to the current working directory, wherever that happens to be

Reply:
Ah, that does sound better. I'll see if I can figure out how to do that!

Reply:
Maybe we shouldn't merge this, and I can instead just make these changes without committing them when working on the static files. It's less than ideal, but so is having `cargo run docs` break if you aren't in the right directory.

Reply:
How about we try an 80-20 solution, since it's just for compiler authors anyway: if the file isn't found, we panic with an error message suggesting that you try running from the root of the roc repository?

It's not amazing ergonomics, but it should suffice for development!

Reply:
I should have asked for help on Zulip first. Ayaz alerted me to the `ROC_WORKSPACE_DIR` env var, which is exactly what I needed! This appears to work regardless of where you run it from!

I refactored a bit, but I'm not at all sure whether my code is good and/or idiomatic. Definitely open to feedback.